File: backend/app/schemas/airflow.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AirflowStatusSchema(BaseModel):
    pipeline_id: str
    dag_id: str
    status: str
    execution_date: datetime | None = None
    last_checked_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class AirflowStatusesResponse(BaseModel):
    statuses: list[AirflowStatusSchema]
    airflow_connected: bool = True

File: frontend/src/types/airflow.ts
export interface AirflowStatus {
  pipeline_id: string;
  dag_id: string;
  status: string;
  execution_date: string | null;
  last_checked_at: string | null;
}

export interface AirflowStatusesResponse {
  statuses: AirflowStatus[];
  airflow_connected: boolean;
}
